### PR TITLE
Simplify `reveal_to_target` logic

### PR DESCRIPTION
**Pull request by evanlinjin**
### Description

- Remove unnecessary check
- Better comments
- Better variable names
- Added another test

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [x] I've added tests for the new feature
* [x] I've added docs for the new feature

### CODE REVIEW

**Comment by 1440000bytes** [comment]
test

File: crates/chain/src/keychain/txout_index.rs [hunk 1]
```diff
@@ -381,40 +381,39 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
...
 
-        // if we can reveal new indexes, the latest revealed index goes here
-        let mut revealed_index = None;
+        debug_assert_eq!(
+            next_reveal_index + lookahead,
+            self.next_store_index(keychain)
+        );
+
+        // if we need to reveal new indexes, the latest revealed index goes here
+        let mut reveal_to_index = None;
```
Comment (vladimirfomene) [inline crates/chain/src/keychain/txout_index.rs:393]: Maybe you should call this `latest_reveal_index`. It might remove the need to have a comment.

**Review by vladimirfomene — APPROVED** [review]
Nice one. This version of the code is easy to follow. I just have a nit.

File: crates/chain/tests/test_keychain_txout_index.rs [hunk 2]
```diff
@@ -210,13 +210,44 @@ fn test_lookahead() {
...
+// - scanning txouts with spks above last stored index
+// expect:
+// - last revealed index should increase as expected
+// - last used index should change as expected
+#[test]
+fn test_scan_with_lookahead() {
+    let (mut txout_index, external_desc, _) = init_txout_index();
+    txout_index.set_lookahead_for_all(10);
+
+    let spks: BTreeMap<u32, Script> = [0, 10, 20, 30]
```
Comment (LLFourn) [inline crates/chain/tests/test_keychain_txout_index.rs:225]: :bulb: after the loop try with 41 and check that it doesn't find it

Comment (evanlinjin) [reply crates/chain/tests/test_keychain_txout_index.rs:225]: Good idea

**Review by LLFourn — APPROVED** [review]

File: crates/chain/src/keychain/txout_index.rs (hunk 1 above)
Comment (evanlinjin) [reply crates/chain/src/keychain/txout_index.rs:393]: I think we probably need a comment anyway to inform that this variable is to be mutated "if we need to reveal new indices".

**Pull request merged**